opa-filter-autoconfigure/src/main/java/com/bisnode/opa/spring/security/filter/autoconfigure/OpaFilterConfigurer.java

```diff
@@ -22,13 +30,24 @@
                 newOpaClient(configuration.getInstance()),
                 configuration.getDocumentPath()
         );
+        RequestMatcher whitelistRequestMatcher = whitelistRequestMatcher(configuration.getEndpointsWhitelist());
 
-        setFilter(new OpaFilter(decider, eventPublisher));
+        setFilter(new OpaFilter(decider, eventPublisher, whitelistRequestMatcher));
     }
 
     private OpaQueryApi newOpaClient(URI instance) {
         return OpaClient.builder()
                 .opaConfiguration(instance.toString())
                 .build();
     }
+
+    private RequestMatcher whitelistRequestMatcher(List<String> whitelistPatterns) {
+        List<RequestMatcher> matchers = Optional.ofNullable(whitelistPatterns)
+                .orElse(List.of()).stream()
```

Comment:
Instead of this line you could probably just write `.flatMap(Collection::stream)`

Reply:
but then I would have to invoke stream mappings in optional mapping and it doesn't look good eg:
`.map(stream -> stream.map(AntPathRequestMatcher::new))
.map(stream -> stream.collect(Collectors.toList()))`